### PR TITLE
[macro_util] Simplify core_reexport

### PR DESCRIPTION
**Pull request by joshlf**

### CODE REVIEW

File: src/macro_util.rs [hunk 1]
```diff
@@ -154,9 +154,7 @@ macro_rules! assert_size_eq {
 }
 
 pub mod core_reexport {
-    pub mod mem {
-        pub use core::mem::transmute;
-    }
+    pub use core::*;
 }
```
Comment (jswrenn) [inline src/macro_util.rs:158]: Instead of all this, why not:
```
pub use ::core as core_reexport;
```

**Review by jswrenn — CHANGES_REQUESTED** [review]

File: src/macro_util.rs (hunk 1 above)
Comment (joshlf) [reply src/macro_util.rs]: Good call, done.

**Review by jswrenn — APPROVED** [review]

**Comment by jswrenn** [comment]
I believe the semver-checks failure is a false-positive, and I've filed a bug report: https://github.com/obi1kenobi/cargo-semver-checks/issues/573

**Comment by joshlf** [comment]
Closed in favor of https://github.com/google/zerocopy/pull/586.

**Pull request closed without merging**